spark/src/main/scala/geotrellis/spark/io/cog/COGLayerReader.scala

```diff
@@ -280,9 +441,8 @@ object COGLayerReader {
                   val tiff = GeoTiffReader[V].read(uri, streaming = true).getOverview(overviewIndex)
                   val map = seq.map { case (gb, sk) => gb -> key.setComponent(sk) }.toMap
 
-                  tiff
-                    .crop(map.keys.toSeq)
-                    .flatMap { case (k, v) => map.get(k).map(i => i -> v) }
+                  readGeoTiff(tiff, map.keys.toSeq)
+                    .flatMap { case (k, v) => map.get(k).map(i => i -> v.asInstanceOf[R]) }
```

Comment:
I think the compiler was still complaining about a type mismatch. I'll look at it again because you're right that shouldn't have to use `asInstanceOf`

Reply:
Locally it compiles without this `asInstanceOf`